### PR TITLE
Btm 311 invoice test

### PR DESCRIPTION
**Pull request by NKannan14**

### CODE REVIEW

File: integration_tests/tests/invoice-athena-tests.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const databaseName = `${prefix}-invoices`;
+    const queryString = `SELECT * FROM "btm_invoices_standardised" ORDER BY invoice_receipt_id ASC, vendor_name asc;`;
+    const queryId = await startQueryExecutionCommand(databaseName, queryString);
+    const queryResults = await formattedQueryResults(queryId);
+    const formattedQueryData = JSON.stringify(queryResults)
+      .replace(/"(\d+)(?:(\.\d*?[1-9]+)0*|\.0*)"/g, '"$1$2"') //removes trailing zeros after decimal point eg 1200.00 to 1200
+      .replace(/"/g, ""); //removes double quotes
+    const response = await getAllObjectsFromS3(bucketName, folderPrefix);
+    const formattedDataFromS3 = JSON.stringify(response)
+      .replace(/(\d+)(?:(\.\d*?[1-9])0*|\.0*)/g, '"$1$2"') //removes trailing zeros after decimal point eg 1200.0 to 1200
```
Comment (markschnitziusgovuk) [inline integration_tests/tests/invoice-athena-tests.ts:39]: This regex to remove trailing zeros looks very similar to the one above, except for a few quotation marks...  Are they the same, and can this be put into a function maybe?

**Review by markschnitziusgovuk — CHANGES_REQUESTED** [review]

File: integration_tests/tests/invoice-athena-tests.ts (hunk 1 above)
Comment (NKannan14) [reply integration_tests/tests/invoice-athena-tests.ts]: moved to function

File: integration_tests/tests/invoice-athena-tests.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { resourcePrefix } from "../helpers/envHelper";
+import path from "path";
+import fs from "fs";
+import { removeTrailingZerosAfterDeciRegExp } from "../helpers/commonHelpers";
+
+const prefix = resourcePrefix();
+
+describe("\nExecute athena query to retrive invoice data and validate it matches with files in standardised-invoice-data s3 bucket\n", () => {
+  test("retrieved invoice details should matches with invoice data in s3 bucket ", async () => {
+    const bucketName = `${resourcePrefix()}-standardised-invoice-data`;
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/tests/invoice-athena-tests.ts:19]: Thanks will change it to storage bucket

File: integration_tests/helpers/commonHelpers.ts [hunk 3]
```diff
@@ -20,4 +20,8 @@ async function waitForTrue(
   });
 }
 
-export { waitForTrue };
+function removeTrailingZerosAfterDeciRegExp(str: string) {
+  return str.replace(/(\d+)(?:(\.\d*?[1-9]+)0*|\.0*)/g, '"$1$2"'); // removes trailing zeros after decimal point eg 1200.00 to 1200
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/helpers/commonHelpers.ts:24]: yes the response from athena query will have number as string whereas from s3 response numbers doesn't have double quotes so when it with regex with "" does not converting the numbers
image

File: integration_tests/tests/invoice-athena-tests.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    const strFromQuery = JSON.stringify(queryResults);
+    const s3Response = await getAllObjectsFromS3(bucketName, folderPrefix);
+    const strFromS3 = JSON.stringify(s3Response);
+
+    const checkStrFromQueryExistsInS3Response = async () => {
+      if (strFromQuery.includes(strFromS3)) {
+        return true;
+      }
+    };
+    expect(checkStrFromQueryExistsInS3Response).toBeTruthy();
```
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/tests/invoice-athena-tests.ts:62]: If i add await shows 
image
for tobeTrue not received return value from function 
image

File: integration_tests/helpers/commonHelpers.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/helpers/commonHelpers.ts]: changed the approach to convert digit to string to presist the decimal point

File: integration_tests/tests/invoice-athena-tests.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (NKannan14) [reply integration_tests/tests/invoice-athena-tests.ts]: just changed the test approach for this test as the function was incorrect

**Review by markschnitziusgovuk — APPROVED** [review]

**Pull request merged**